test/e2e/cross_os/test_job_submissions.py

```diff
@@ -370,3 +372,57 @@ def test_worker_run_with_number_of_environments(
         ), "Expected number of Hello statements not found in job logs."
 
         assert job.task_run_status == TaskStatus.SUCCEEDED
+
+    def test_worker_streams_logs_to_cloudwatch(
+        self,
+        deadline_resources: DeadlineResources,
+        deadline_client: DeadlineClient,
+        worker: EC2InstanceWorker,
+    ) -> None:
+
+        job = Job.submit(
+            client=deadline_client,
+            farm=deadline_resources.farm,
+            queue=deadline_resources.queue_a,
+            priority=98,
+            template={
+                "specificationVersion": "jobtemplate-2023-09",
+                "name": "Hello World Job",
+                "steps": [
+                    {
+                        "name": "Step0",
+                        "script": {
+                            "actions": {"onRun": {"command": "echo", "args": ["HelloWorld"]}}
+                        },
+                    },
+                ],
+            },
+        )
+
+        job.wait_until_complete(client=deadline_client)
+        logs_client: boto3.client = boto3.client(
+            "logs",
+            config=botocore.config.Config(retries={"max_attempts": 10, "mode": "adaptive"}),
+        )
+
+        # Retrieve job output and verify the echo is printed
+        job_logs = job.get_logs(deadline_client=deadline_client, logs_client=logs_client)
+        full_log: str = "\n".join(
+            [le.message for _, log_events in job_logs.logs.items() for le in log_events]
+        )
+
+        assert (
+            full_log.count("HelloWorld") == 1
+        ), "Expected number of HelloWorld statements not found in job logs."
+
+        # Retrieve worker logs and verify that it's not empty
+        worker_log_group_name: str = (
+            f"/aws/deadline/{deadline_resources.farm.id}/{deadline_resources.fleet.id}"
+        )
+        worker_id = worker.worker_id
+
+        worker_logs = logs_client.get_log_events(
```

Comment:
Since this is a recycled log stream from a persisted resource, is this 100% correct?

Should we verify the logs exist AFTER the jobs are created?

Reply:
You're right. I missed that part. I'll change it so that we verify logs exist after the job is created